pkg/boot/crawl/crawl.go

```diff
@@ -105,34 +105,74 @@ func (c *Crawler) FindPeers(ctx context.Context, ns string, opt ...discovery.Opt
 	go func() {
 		defer cancel()
 
-		var addr net.UDPAddr
-		for c.active(ctx) && iter.Next(&addr) {
```

Comment:
Isn't it possible to add a check of `context.Canceled, context.DeadlineExceeded` here? 

This way we would avoid all the extra logic right? The current implementation is quite readable. And  if we avoided an extra struct (`rangeIter`) It is much readable in my opinion.

Reply:
Good thought. I'll push a fix asap.